In [2]:
import pandas as pd

In [4]:
import numpy as np

In [5]:
file = 'Toda_201911.txt'

num_features = 10

feature_cols = [str(f_index) for f_index in range(1, num_features+1)]

df = pd.read_csv(file, sep=" ", header=None)
df.drop(columns=df.columns[[1,2,3,5,7,9,11,13,15,17,19,21]], axis=1, inplace=True)  # remove redundant keys
#print(df)
#print(self.num_features, len(df.columns))
#assert self.num_features == len(df.columns) - 2

#for c in range(1, self.num_features + 2):  # remove keys per column from key:value
#    df.iloc[:, c] = df.iloc[:, c].apply(lambda x: x.split(":")[1])

df.columns = ['rele_truth', 'teiban', 'player_age', 'player_weight', 'player_level', 'all_winrate', 'all_2rate', 
          'local_winrate', 'local_2rate', 'motor_2rate', 'boat_2rate'] #+ feature_cols #+ ['#docid', 'inc', 'prob']

#if self.data_id in MSLETOR_SEMI and self.data_dict['unknown_as_zero']:
#    self.df[self.df[self.feature_cols] < 0] = 0

for c in ['rele_truth', 'teiban', 'player_age', 'player_weight', 'player_level', 'all_winrate', 'all_2rate', 
          'local_winrate', 'local_2rate', 'motor_2rate', 'boat_2rate']:# + feature_cols:
    df[c] = df[c].astype(np.float32)

#df['rele_binary'] = (df['rele_truth'] > 0).astype(np.float32)  # additional binarized column for later filtering

print(pd.DataFrame(df))

      rele_truth  teiban  player_age  player_weight  player_level  \
0            3.0     1.0        26.0           50.0           3.0   
1            2.0     2.0        55.0           50.0           2.0   
2            1.0     3.0        57.0           50.0           2.0   
3            4.0     4.0        42.0           47.0           1.0   
4            5.0     5.0        26.0           48.0           2.0   
5            0.0     6.0        22.0           44.0           2.0   
6            5.0     1.0        40.0           52.0           2.0   
7            2.0     2.0        47.0           53.0           2.0   
8            0.0     3.0        25.0           52.0           2.0   
9            4.0     4.0        40.0           53.0           0.0   
10           1.0     5.0        46.0           58.0           2.0   
11           3.0     6.0        31.0           52.0           2.0   
12           3.0     1.0        53.0           53.0           2.0   
13           5.0     2.0        28

In [6]:
df.head(10)

rele_truth  teiban  player_age  player_weight  player_level  all_winrate  \
0         3.0     1.0        26.0           50.0           3.0         1.56   
1         2.0     2.0        55.0           50.0           2.0         3.52   
2         1.0     3.0        57.0           50.0           2.0         3.24   
3         4.0     4.0        42.0           47.0           1.0         6.88   
4         5.0     5.0        26.0           48.0           2.0         4.52   
5         0.0     6.0        22.0           44.0           2.0         2.54   
6         5.0     1.0        40.0           52.0           2.0         4.59   
7         2.0     2.0        47.0           53.0           2.0         4.87   
8         0.0     3.0        25.0           52.0           2.0         4.45   
9         4.0     4.0        40.0           53.0           0.0         6.74   

   all_2rate  local_winrate  local_2rate  motor_2rate  boat_2rate  
0   2.780000           2.00     5.260000    33.330002   28.330000  
1   9.640000           4.05    17.070000    34.380001   30.360001  
2   8.060000           4.39    16.670000    34.330002   32.259998  
3  52.939999           0.00     0.000000    36.669998   26.670000  
4  21.540001           4.38    37.500000    33.900002   40.349998  
5   5.050000           2.63     6.670000    36.990002   54.840000  
6  20.000000           5.18    28.129999    33.330002   38.180000  
7  26.770000           5.31    36.360001    37.700001   34.779999  
8  24.420000           4.29    25.000000    36.919998   35.820000  
9  53.189999           6.20    51.220001    48.529999   35.209999

In [7]:
dfs = df.iloc[:, 1:].apply(lambda x: (x-x.mean()), axis=0)
dfs.head(10)

teiban  player_age  player_weight  player_level  all_winrate  all_2rate  \
0    -2.5   -12.25383      -1.914749      1.701149    -3.789445 -31.471178   
1    -1.5    16.74617      -1.914749      0.701149    -1.829445 -24.611179   
2    -0.5    18.74617      -1.914749      0.701149    -2.109445 -26.191177   
3     0.5     3.74617      -4.914749     -0.298851     1.530555  18.688820   
4     1.5   -12.25383      -3.914749      0.701149    -0.829445 -12.711178   
5     2.5   -16.25383      -7.914749      0.701149    -2.809445 -29.201180   
6    -2.5     1.74617       0.085251      0.701149    -0.759445 -14.251179   
7    -1.5     8.74617       1.085251      0.701149    -0.479445  -7.481178   
8    -0.5   -13.25383       0.085251      0.701149    -0.899445  -9.831179   
9     0.5     1.74617       1.085251     -1.298851     1.390555  18.938820   

   local_winrate  local_2rate  motor_2rate  boat_2rate  
0      -3.229895   -28.754023    -0.254242   -6.591112  
1      -1.179894   -16.944023     0.795757   -4.561111  
2      -0.839895   -17.344023     0.745758   -2.661114  
3      -5.229895   -34.014023     3.085754   -8.251112  
4      -0.849895     3.485977     0.315758    5.428886  
5      -2.599895   -27.344023     3.405758   19.918888  
6      -0.049895    -5.884024    -0.254242    3.258888  
7       0.080105     2.345978     4.115757   -0.141113  
8      -0.939895    -9.014023     3.335754    0.898888  
9       0.970105    17.205978    14.945755    0.288887

In [8]:
from sklearn.linear_model import LinearRegression

In [11]:
model_lr_std = LinearRegression()
model_lr_std.fit(dfs, df.rele_truth)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [13]:
print(model_lr_std.coef_)
print(model_lr_std.intercept_)
print(model_lr_std.score(dfs, df.rele_truth))

[-0.3136385   0.00035617  0.01678238 -0.1296297   0.0086118   0.00559373
 -0.00075812 -0.00039992  0.01233556  0.00615345]
2.5
0.11930384311114184


									

|teiban|player_age|player_weight|player_level|all_winrate|all_2rate|local_winrate|local_2rate|motor_2rate|boat_2rate|
|---|---|---|---|---|---|---|---|---|---|
|-0.3136385|0.00035617 | 0.01678238 | -0.1296297 |0.0086118|0.00559373|-0.00075812| -0.00039992|0.01233556|0.00615345|

In [15]:
model_lr_std.predict(dfs)

array([2.7886567, 2.689396 , 2.376197 , ..., 2.4517438, 2.3028264,
       1.8209693], dtype=float32)